lib/console.js

```diff
@@ -454,10 +450,38 @@ Console.prototype.table = function(tabularData, properties) {
   return final(keys, values);
 };
 
-module.exports = new Console({
+function noop() {}
+
+// See https://console.spec.whatwg.org/#console-namespace
+// > For historical web-compatibility reasons, the namespace object
+// > for console must have as its [[Prototype]] an empty object,
+// > created as if by ObjectCreate(%ObjectPrototype%),
+// > instead of %ObjectPrototype%.
+
+// Since in Node.js, the Console constructor has been exposed through
+// require('console'), we need to keep the Console constructor but
+// we cannot actually use `new Console` to construct the global console.
+// Therefore, the console.Console.prototype is not
+// in the global console prototype chain anymore.
+const consolePrototype = Object.create(Object.prototype);
+const globalConsole = Object.create(consolePrototype);
+ConsoleImpl.call(globalConsole, {
   stdout: process.stdout,
   stderr: process.stderr
 });
-module.exports.Console = Console;
 
-function noop() {}
+function Console(...args) {
+  if (!(this instanceof Console)) {
```

Comment:
instead of `!(this instanceof Console)` could use `!new.target`

Reply:
Not that anybody does that, but these would behave differently if somebody did ES5-style inheritance from this class, right? The current version is also a bit more idiomatic, imo (same for the comment above…)